### PR TITLE
GH1487: made the nocache flag as opt in

### PR DESCRIPTION
**Pull request by pitermarx**

### CODE REVIEW

File: src/Cake.NuGet/NuGetPackageInstaller.cs [hunk 1]
```diff
@@ -228,7 +228,13 @@ private FilePath GetNuGetPath()
                 arguments.Append("-Prerelease");
             }
 
-            arguments.Append("-ExcludeVersion -NonInteractive -NoCache");
+            // NoCache
+            if (definition.Parameters.ContainsKey("nocache"))
```
Comment (patriksvensson) [inline src/Cake.NuGet/NuGetPackageInstaller.cs]: Would be nice if you could add unit tests for this (to make sure it's only included when it's set).

Comment (pitermarx) [inline src/Cake.NuGet/NuGetPackageInstaller.cs]: can you sugest where? I'm having a hard time navigating your test codebase! 
SO MANY TESTS!

Comment (gep13) [inline src/Cake.NuGet/NuGetPackageInstaller.cs]: @pitermarx somewhere around here:

http://sourcebrowser.io/Browse/cake-build/cake/src/Cake.NuGet.Tests/Unit/NuGetPackageInstallerTests.cs#166

I would think would make sense.

**Comment by pitermarx** [comment]
added unit tests

**Comment by gep13** [comment]
@pitermarx is there an issue associated with this PR?

**Comment by pitermarx** [comment]
Yes https://github.com/cake-build/cake/issues/1487

**Comment by gep13** [comment]
@pitermarx thank you for making this change!

**Comment by pitermarx** [comment]
Thank you for merging.

**Pull request merged**